src/data/set/function.lean

```diff
@@ -261,6 +261,9 @@ subtype.map f h
 @[simp] lemma maps_to.coe_restrict_apply (h : maps_to f s t) (x : s) :
   (h.restrict f s t x : β) = f x := rfl
 
+lemma cod_restrict_restrict (h : maps_to f s t) :
+  cod_restrict (s.restrict f) t (λ x, h x.2) = h.restrict f s t := rfl
```

Comment:
Can you add a primed version with `h` a free variable on the LHS` instead?

Reply:
Do you mean a version where the assumption is `(h : ∀ (x : s), f x.2 ∈ t`?

Reply:
Yes, exactly